src/vos/vos_io.c

```diff
@@ -578,6 +578,13 @@ akey_fetch(struct vos_io_context *ioc, daos_handle_t ak_toh)
 	for (i = 0; i < iod->iod_nr; i++) {
 		daos_size_t rsize;
 
+		if (iod->iod_recxs[i].rx_nr == 0) {
+			D_ERROR("Invalid read IOD at %d: idx %lu, nr %lu\n", i,
+				(unsigned long)iod->iod_recxs[i].rx_idx,
+				(unsigned long)iod->iod_recxs[i].rx_nr);
+			D_GOTO(out, rc = -DER_INVAL);
```

Comment:
in EC obj IO, when there are multiple iods and each iod with different targets, then the dispatched IO request to some targets maybe with iod_nr == 0. Maybe can just return 0 for this case?  or I can check the case before calling to VOS, both are fine.

@mchaarawi  On the other hand, seems dc_array possible with zero iod_nr?

Reply:
Ah, sorry, zero iod_nr is a different case, here is to handle zero rx_nr, please just ignore my above comments.